test/common/stats/thread_local_store_test.cc

```diff
@@ -189,6 +260,152 @@ TEST_F(StatsThreadLocalStoreTest, BasicScope) {
   // Includes overflow stat.
   EXPECT_CALL(*this, free(_)).Times(5);
 }
+TEST_F(StatsThreadLocalStoreTest, BasicSingleHistogramMerge) {
+  InSequence s;
+  store_->initializeThreading(main_thread_dispatcher_, tls_);
+
+  Histogram& h1 = store_->histogram("h1");
+  EXPECT_EQ("h1", h1.name());
+
+  expectCallAndAccumulate(h1, 0);
+  expectCallAndAccumulate(h1, 43);
+  expectCallAndAccumulate(h1, 41);
+  expectCallAndAccumulate(h1, 415);
+  expectCallAndAccumulate(h1, 2201);
+  expectCallAndAccumulate(h1, 3201);
+  expectCallAndAccumulate(h1, 125);
+  expectCallAndAccumulate(h1, 13);
+
+  EXPECT_EQ(1, validateMerge());
+
+  store_->shutdownThreading();
+  tls_.shutdownThread();
+
+  // Includes overflow stat.
+  EXPECT_CALL(*this, free(_));
+}
+
+TEST_F(StatsThreadLocalStoreTest, BasicMultiHistogramMerge) {
+  InSequence s;
+  store_->initializeThreading(main_thread_dispatcher_, tls_);
+
+  Histogram& h1 = store_->histogram("h1");
+  Histogram& h2 = store_->histogram("h2");
+  EXPECT_EQ("h1", h1.name());
+  EXPECT_EQ("h2", h2.name());
+
+  expectCallAndAccumulate(h1, 1);
+  expectCallAndAccumulate(h2, 1);
+  expectCallAndAccumulate(h2, 2);
+
+  EXPECT_EQ(2, validateMerge());
+
+  store_->shutdownThreading();
```

Comment:
Can you factor out the closing and opening stanzas into SetUp and TearDown?

If this isn't applicable to all tests in the file, you can subclass your test method, or have explicit test-class methods which you can call from the relevant tests.

Reply:
I tried to do opening stanzas in to SetUp but since I need the set up histograms in subsequent calls, I could not make them.

Reply:
I was just thinking about calling initializeThreading() in SetUp(), and putting InSequence in the class object (or maybe a histogram-related subclass.  I wasn't thinking about initiating the histograms themselves as they are not all common.

But for TearDown, can't you do all these 3 lines?  They seem pretty common.  I'm not sure about the EXPECT_CALL below...could that be done in TearDown, or in the test body, which would re-order it?

Reply:
Yes. That can be done. I have added those functions.